android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java

```diff
@@ -243,56 +256,67 @@ public MatchingEngineHelper(final Builder builder) {
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyMaxNumUpdates);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestPort);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyThreshold);
+        onEdgeEventPreferenceChanged(prefs, prefKeyPerfMarginSwitch);
+        onEdgeEventPreferenceChanged(prefs, prefKeyAutoMigration);
     }
 
-    public void startEdgeEvents() {
+    public void startEdgeEvents(boolean newConnection) {
+        if (!mEdgeEventsEnabled) {
+            Log.e(TAG, "Aborting attempt to startEdgeEvents while mEdgeEventsEnabled="+ false);
+            return;
+        }
         if (mClosestCloudlet == null) {
             Log.w(TAG, "Must perform findCloudlet before starting Edge Events.");
             return;
         }
-        mEdgeEventsConfig = me.createDefaultEdgeEventsConfig();
-        Log.i(TAG, "Default EdgeEventsConfig="+mEdgeEventsConfig);
-
-        if (mOverrideDefaultConfig) {
-            // In this case, all attributes for mEdgeEventsConfig will be set from Preferences.
-            // Otherwise, we retain the default values.
-            initEdgeEventsConfigPrefs();
-            Log.i(TAG, "Custom EdgeEventsConfig="+mEdgeEventsConfig);
-        }
 
         new Thread(() -> {
+            mEdgeEventsConfig = me.createDefaultEdgeEventsConfig();
+            String message = "Using Default EdgeEventsConfig=" + mEdgeEventsConfig;
+            Log.i(TAG, message);
+
+            if (mOverrideDefaultConfig) {
+                // In this case, all attributes for mEdgeEventsConfig will be set from Preferences.
+                // Otherwise, we retain the default values.
+                initEdgeEventsConfigPrefs();
+                message = "Using Custom EdgeEventsConfig="+mEdgeEventsConfig;
+                Log.i(TAG, message);
+            }
+            meHelperInterface.showMessage(message);
+
             // Only do this once.
             if (mEdgeEventsSubscriber == null) {
                 mEdgeEventsSubscriber = new EdgeEventsSubscriber();
                 me.getEdgeEventsBus().register(mEdgeEventsSubscriber);
             }
 
-            Log.i(TAG, "mEdgeEventsConfig="+mEdgeEventsConfig.toString());
-            meHelperInterface.showMessage("mEdgeEventsConfig="+mEdgeEventsConfig.toString());
-
-            String message;
-            if (me.isAutoMigrateEdgeEventsConnection()) {
-                message = "Auto-migrating to new app inst";
-            } else {
-                if (mEdgeEventsRunning) {
+            if (mEdgeEventsRunning) {
+                if (me.isAutoMigrateEdgeEventsConnection()) {
+                    if (newConnection) {
+                        message = "Auto-migrating to new app inst";
+                    } else {
+                        message = "Retaining connection to existing app inst";
+                    }
+                } else {
                     message = "Restarting ServerEdgeEvents";
                     me.stopEdgeEvents();

```

Comment:
You can also do one of the following in MatchingEngine:
    public CompletableFuture<Boolean> restartEdgeEventsFuture();
    synchronized public boolean restartEdgeEvents(); // also an alias wrapper "switchedToNextCloudlet"

Reply:
@lgarner Does `restartEdgeEvents` reuse the same config? This stop/start code is passing a newly constructed mEdgeEventsConfig, which starts as the default, then gets modified according to user preferences. Is there a way to use "restart" and pass in the new config?

Reply:
Hm, important point. No, you'd need a new startEdgeEvents(config) to use a new config. Restart is sort of a convenience thing. It just means restart, not restart with a new config. Since it's app controlled, there is some optimizations in restart already.